src/ui/screens/ImportAccountWithJsonFileScreen.tsx

```diff
@@ -24,28 +23,22 @@ export function ImportAccountWithJsonFileScreen({navigation}: {navigation: Navig
   const parsedJson = jsonContent ? tryParseJson(jsonContent) : undefined;
   const [isPasswordVisible, setIsPasswordVisible] = React.useState(false);
   const [password, setPassword] = React.useState('');
-  const {addAccount} = useAccounts();
 
-  async function restoreAccount() {
+  const {restoreAccount} = useKeyring();
+
+  async function onRestoreAccount() {
     if (parsedJson && password) {
       try {
-        // TODO: NOT WORKING WITH POLKADOT EXTENSION THE ENCODING IS DIFFERENT THERE
-        await SubstrateSign.decryptData(parsedJson.encoded, password);
-
-        const newAcc = {
-          address: parsedJson.address,
-          encoded: parsedJson.encoded,
-          meta: {...parsedJson.meta, network: currentNetwork.key, isFavorite: false},
-          isExternal: false,
-        };
-        addAccount(newAcc);
-        SecureKeychain.setPasswordByServiceId(password, 'BIOMETRICS', parsedJson.address);
-
+        const account = await restoreAccount({
+          json: parsedJson,
+          password,
+          network: currentNetwork.key,
+        });
+        SecureKeychain.setPasswordByServiceId(password, 'BIOMETRICS', account.address as string);
```

Comment:
What is the type of `account.address` ?

Reply:
unknown, this is coming from the WebView.